src/app.js

```diff
@@ -0,0 +1,194 @@
+    e.preventDefault();
+}
+countryInformation.addEventListener('click', preventDefault)
+
+//Puts the searchbar in the countryInformation form.
+countryInformation.appendChild(searchBar);
+
+// Makes identifier variables of the above created submit and searchbar.
+const submitSearch = document.getElementById('submitSearch');
+const searchText = document.getElementById('searchText');
+
+
+// The async function that fetches data through an api.
+async function getCountries() {
+    try {
+
+        //Gets data
+        const result = await axios.get('https://restcountries.com/v2/all');
+
+        // Puts the array with objects we got in a variable
+        const allData = result.data;
+
+        //An indexNumber to identify a specific object in this case a country!
+        let indexNumber;
+
+
+        // Function that is being called when user puts something in the search field!
+        // If the user input exists it will connect the input to the index number, identifying the object array[indexNumber].
+        const search = (e) => {
+           // returns the names of the objects
+            const names = allData.map((item) => {
+              return item.name
+            })
+            //Checks if the input name exists in the array object, if it does then it returns the indexNumber of the array where the object is located in.
+            if (names.includes(e.target.value)) {
+                   for (let i = 0; i < names.length; i++) {
+                       if (names[i] === e.target.value) {
+                           indexNumber = i;
+                       }
+                   }
+               }
+
+               //If the name does not exist it will return false!
+               else {
+                   indexNumber = false;
+               }
+        }
+
+
+        // Makes a div where the country name and flag will be located in!
+        const flagAndTitle = document.createElement('div');
+        flagAndTitle.setAttribute('style', 'display: flex; align-items: center;' +
+            'justify-content: center;');
+
+        // Makes a div where the country description will be located in!
+        const countryDescription = document.createElement('div');
+
+
+        // A function that pushes data in the flagAndTitle and countryDescription divs after submitting!
+const printCountry = () => {
+    //If an indexNumber is connected it will play everything inside this if statement.
+    if (indexNumber === 0 || indexNumber > 0 && indexNumber < allData.length) {
+
+        // The languages variable checks whether a country has 1 or multiple languages!
+        //If a country has multiple languages it will make it look good in the div!
+        const languages = allData[indexNumber].languages.map((item) => {
+            return item.name
+        })
+        for (let i = 1; i < languages.length; i++) {
+            if (i < languages.length - 1) {
+                languages[0] = languages[0] + ', ' + languages[i];

```

Comment:
Het is een badpractice om een array aan te passen waar je overheen loopt. Ik zou een aparte variabele maken om de string mee op te bouwen.